presidio-analyzer/presidio_analyzer/predefined_recognizers/azure_ai_language.py

```diff
@@ -0,0 +1,155 @@
+            ta_client = self.__authenticate_client(azure_ai_key, azure_ai_endpoint)
+        self.ta_client = ta_client
+
+    @staticmethod
+    def get_azure_ai_supported_entities() -> List[str]:
+        """Return the list of all supported entities for Azure AI Language."""
+        from azure.ai.textanalytics._models import PiiEntityCategory
+
+        return [r.value.upper() for r in PiiEntityCategory]
+
+    @staticmethod
+    def __authenticate_client(key: str, endpoint: str) -> TextAnalyticsClient:
+        """Authenticate the client using the key and endpoint."""
+
+        ta_credential = AzureKeyCredential(key)
+        text_analytics_client = TextAnalyticsClient(
+            endpoint=endpoint, credential=ta_credential
+        )
+        return text_analytics_client
+
+    def analyze(
+        self, text: str, entities: List[str] = None, nlp_artifacts: NlpArtifacts = None
+    ) -> List[RecognizerResult]:
+        """
+        Analyze text using Azure AI Language.
+
+        :param text: Text to analyze
+        :param entities: List of entities to return
+        :param nlp_artifacts: Object of type NlpArtifacts, not used in this recognizer.
+        :return: A list of RecognizerResult, one per each entity found in the text.
+        """
+        if not entities:
+            entities = self.supported_entities
+        response = self.ta_client.recognize_pii_entities(
+            [text], language=self.supported_language
+        )
+        results = [doc for doc in response if not doc.is_error]
+        recognizer_results = []
+        for res in results:
+            for entity in res.entities:
+                entity.category = entity.category.upper()
+                if entity.category not in self.supported_entities:
+                    continue
+                if entity.category not in entities:
+                    continue
+                analysis_explanation = AzureAILanguageRecognizer._build_explanation(
+                    original_score=entity.confidence_score,
+                    entity_type=entity.category,
+                )
+                recognizer_results.append(
+                    RecognizerResult(
+                        entity_type=entity.category,
+                        start=entity.offset,
+                        end=entity.offset + len(entity.text),
+                        score=entity.confidence_score,
+                        analysis_explanation=analysis_explanation,
+                    )
+                )
+
+        return recognizer_results
+
+    @staticmethod
+    def _build_explanation(
+        original_score: float, entity_type: str
+    ) -> AnalysisExplanation:
+        explanation = AnalysisExplanation(
+            recognizer=AzureAILanguageRecognizer.__class__.__name__,
+            original_score=original_score,
+            textual_explanation=f"Identified as {entity_type} by Azure AI Language",
+        )
+        return explanation
+
+    def load(self) -> None:
```

Comment:
redundant method as the super class does the same?

Reply:
Parent has an abstract load method. BTW I didn't do it but I guess we can go with an implementation of a RemoteRecognizer instead of an EntityRecognizer. WDYT?

Reply:
Yes, remote like the old one?

Reply:
Yes

Reply:
Fixed